crates/linker/src/lib.rs

```diff
@@ -35,6 +35,7 @@ pub fn supported(link_type: LinkType, target: Target) -> bool {
             Target::WinX64 => true,
             // macho support is incomplete
             Target::MacX64 => false,
+            Target::MacArm64 => true,
```

Comment:
I am not 100% sure if this is wanted here (aka won't break users' programs). If so, I will revert it back to `false`. I was also thinking that perhaps it could be feature-gated until the linker is ready for some prime time?

Reply:
Yeah, I think this should remain false until the linker is ready to be used by default. So at least all of the cli tests successfully link with the macho surgical linker.

Reply:
Yeah, an env var sounds better. I will follow up with this, thanks for the suggestion!